src/swap/SwapPending.tsx

```diff
@@ -0,0 +1,154 @@
+  }
+
+  const navigateToReviewScreen = () => {
+    navigate(Screens.SwapReviewScreen)
+  }
+
+  const navigateToSupport = () => {
+    dispatch(swapReset())
+    navigate(Screens.SwapScreen)
+    navigate(Screens.SupportContact)
+  }
+
+  const swapDisplay = useMemo(() => {
+    switch (swapState) {
+      default:
+      case SwapState.START:
+        return (
+          <>
+            <Text style={styles.text}>{t('swapCompleteScreen.swapPending')}</Text>
+            <Text style={styles.subText}>{t('swapCompleteScreen.exchangeRateSubtext')}</Text>
+          </>
+        )
+      case SwapState.APPROVE:
+        return (
+          <>
+            <Text style={styles.text}>{t('swapCompleteScreen.swapPending')}</Text>
+            <Text style={styles.subText}>{t('swapCompleteScreen.approvingSubtext')}</Text>
+          </>
+        )
+      case SwapState.EXECUTE:
+        return (
+          <>
+            <Text style={styles.text}>{t('swapCompleteScreen.swapPending')}</Text>
+            <Text style={styles.subText}>{t('swapCompleteScreen.completingSubtext')}</Text>
+          </>
+        )
+      case SwapState.ERROR:
+        return (
+          <Dialog
+            isVisible={true}
+            title={t('swapCompleteScreen.swapErrorModal.title')}
+            actionText={t('swapCompleteScreen.swapErrorModal.action1')}
+            actionPress={navigateToSwapStart}
+            secondaryActionText={t('swapCompleteScreen.swapErrorModal.action2')}
+            secondaryActionPress={navigateToSupport}
+          >
+            {t('swapCompleteScreen.swapErrorModal.body')}
+          </Dialog>
+        )
+      case SwapState.PRICE_CHANGE:
+        return (
+          <Dialog
+            isVisible={true}
+            title={t('swapCompleteScreen.swapPriceModal.title')}
+            actionText={t('swapCompleteScreen.swapPriceModal.action')}
+            actionPress={navigateToReviewScreen}
+            testID="PriceChangeModal"
+            onBackgroundPress={navigateToReviewScreen}
+          >
+            {t('swapCompleteScreen.swapPriceModal.body1')}
+            {'\n\n'}
+            {t('swapCompleteScreen.swapPriceModal.body2')}
+          </Dialog>
+        )
+      // These states are the first and last of the swap flow
+      // Avoids a flicker when resetting state.swap.swapInfo
+      case SwapState.USER_INPUT:
+      case SwapState.COMPLETE:
+        return (
+          <>
+            <Text style={styles.text}>{t('swapCompleteScreen.swapSuccess')}</Text>
+            <Button text={t('swapCompleteScreen.swapAgain')} onPress={navigateToSwapStart} />
+          </>
+        )
+    }
+  }, [swapState])
+
+  // Prevent back button on Android
+  useEffect(() => {
+    const backPressListener = () => true
+    BackHandler.addEventListener('hardwareBackPress', backPressListener)
+    return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
+  }, [])
+
+  return (
+    <SafeAreaView style={{ flex: 1 }}>
```

Comment:
move this style to `styles`? :)

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2891/commits/00b9f8d7b0e8b57b7de7f174c1899dcc64b57b1d!